### PR TITLE
Added relations support to entity mapping generator

### PR DESCRIPTION
**Pull request by zeopix**

### CODE REVIEW

File: Command/AutoComplete/EntitiesAutoCompleter.php [hunk 1]
```diff
@@ -27,25 +31,34 @@ public function __construct(EntityManagerInterface $manager)
...
-
         $entities = $configuration
             ->getMetadataDriverImpl()
             ->getAllClassNames()
         ;
 
-        return array_map(function ($entity) use ($namespaceReplacements) {
-            return strtr($entity, $namespaceReplacements);
-        }, $entities);
+        if ($applyNamespaceReplacements == true) {
```
Comment (phansys) [inline Command/AutoComplete/EntitiesAutoCompleter.php]: Please, use Yoda condition and strict comparison.

File: Command/GenerateDoctrineEntityCommand.php [hunk 2]
```diff
@@ -216,9 +218,28 @@ private function addFields(InputInterface $input, OutputInterface $output, Quest
         }
         $output->writeln('');
 
-        $fieldValidator = function ($type) use ($types) {
+        $output->write('<info>Available relations:</info> ');
+
+        $relations = array('many_to_many', 'many_to_one', 'one_to_one', 'one_to_many');
+        foreach ($relations as $i => $relation) {
+            $count += strlen($relation);
```
Comment (phansys) [inline Command/GenerateDoctrineEntityCommand.php]: What is the indended use for this? You're counting how many chars have the relationship's names and this var seems to be unused.

File: Command/GenerateDoctrineEntityCommand.php [hunk 3]
```diff
@@ -216,9 +218,28 @@ private function addFields(InputInterface $input, OutputInterface $output, Quest
...
+            $output->write(sprintf('<comment>%s</comment>', $relation));
+            if (count($relations) != $i + 1) {
```
Comment (phansys) [inline Command/GenerateDoctrineEntityCommand.php]: Shouldn't be easier something like ```$output->write('<comment>'.implode('</comment>, <comment>', $relations).'</comment>.';```

File: Command/GenerateDoctrineEntityCommand.php [hunk 4]
```diff
@@ -289,6 +310,88 @@ private function addFields(InputInterface $input, OutputInterface $output, Quest
...
             }
 
+            if ($type == 'one_to_many') {
+                $question = new Question($questionHelper->getQuestion('Target Entity class', ''), '');
+                $question->setAutocompleterValues($autocompleteEntities);
+                $data['targetEntity'] = $questionHelper->ask($input, $output, $question);
+
+                $question = new Question($questionHelper->getQuestion('Mapped by', ''), '');
+                $mappedBy = $questionHelper->ask($input, $output, $question);
+                if (empty($mappedBy) == false)
```
Comment (phansys) [inline Command/GenerateDoctrineEntityCommand.php]: Please, use braces.

File: Command/GenerateDoctrineEntityCommand.php [hunk 5]
```diff
@@ -289,6 +310,88 @@ private function addFields(InputInterface $input, OutputInterface $output, Quest
                 $data['length'] = $questionHelper->ask($input, $output, $question);
             }
 
+            if ($type == 'one_to_many') {
+                $question = new Question($questionHelper->getQuestion('Target Entity class', ''), '');
+                $question->setAutocompleterValues($autocompleteEntities);
```
Comment (phansys) [inline Command/GenerateDoctrineEntityCommand.php]: The user input should be validated.

File: Command/GenerateDoctrineEntityCommand.php [hunk 6]
```diff
@@ -289,6 +310,88 @@ private function addFields(InputInterface $input, OutputInterface $output, Quest
...
+                    if (empty($inversedBy) == false)
+                        $data['inversedBy'] = $inversedBy;
+
+                    $question = new Question($questionHelper->getQuestion('Join Table Name', ''), '');
+                    $joinTableName = $questionHelper->ask($input, $output, $question);
+                    if (empty($joinTableName) == false)
+                        $data['joinTable'] = array('name' => $joinTableName);
+
+                    $data['isOwningSide'] = true;
+                }else{
```
Comment (phansys) [inline Command/GenerateDoctrineEntityCommand.php]: Please, respect [PSR-2 CS](https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide.md#51-if-elseif-else).

File: Command/GenerateDoctrineEntityCommand.php (hunk 4 above)
Comment (phansys) [inline Command/GenerateDoctrineEntityCommand.php]: You should do just a strict comparison of the user's input against the default value.

**Comment by phansys** [comment]
These options should be supported in ```--no-interactive``` mode and some tests are worth.

**Comment by phansys** [comment]
Closes #10, #278.

**Comment by javiereguiluz** [comment]
Recently we decided to refocus the purpose of this bundle and work only on its core features. Supporting entity associations is one of the features that we prefer to not implement yet. The alternative solution is to use third-party bundles such as SonataAdmin, which support this and many other complex features.

Therefore, I'm closing this and all the other related issues and pull requests as "won't fix". Thanks for understanding it.

**Comment by ryall** [comment]
@javiereguiluz Is there a reason you don't consider entity associations a core feature when they are used in a high percentage of all entity classes?

**Pull request closed without merging**